standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/Warehouse.java

```diff
@@ -240,15 +240,15 @@ public Path determineDatabasePath(Catalog cat, Database db) throws MetaException
       if (db.getName().equalsIgnoreCase(DEFAULT_DATABASE_NAME)) {
         return getWhRootExternal();
       } else {
-        return new Path(getWhRootExternal(), dbDirFromDbName(db));
+        return new Path(getWhRootExternal(), dbDirFromDbName(db.getName()));
       }
     } else {
-      return new Path(getDnsPath(new Path(cat.getLocationUri())), dbDirFromDbName(db));
+      return new Path(getDnsPath(new Path(cat.getLocationUri())), dbDirFromDbName(db.getName()));
     }
   }
 
-  private String dbDirFromDbName(Database db) throws MetaException {
-    return db.getName().toLowerCase() + DATABASE_WAREHOUSE_SUFFIX;
+  private String dbDirFromDbName(final String dbName) {
+    return MetaStoreUtils.encodeTableName(dbName.toLowerCase()) + DATABASE_WAREHOUSE_SUFFIX;
```

Comment:
I would suggest to use `org.apache.hadoop.hive.metastore.utils.StringUtils.normalizeIdentifier ` it not just do the lowerCase but also remove the whitespaces (if somebody is pervert to use it :) ) As I see mostly the project use this util method for this.  

I know the previous implementation also not used that

Reply:
Yes, that sounds like a reasonable suggestion to me, although I hope that nobody is pervert enough use a db name with trailing spaces ;)

Reply:
Haha you can't imagine! But there were even more pervert characters that I have seen :D